### ISSUE TITLE
ObservationStats from_observation fails if background is zero

### ISSUE THREAD

**Issue by lmohrmann** [gammapy/gammapy#1955]
After fdc0e9b2e22b205353f6ef8f29baf548b81890df, `ObservationStats.from_observation` gives nonsense results if `bg_estimate.a_off` equals zero. The problem is that `a_off` is set to 1 in the constructor if `alpha` is not `None`:
```
if alpha is not None:
            self.a_on = alpha
            self.a_off = 1
            self.alpha_obs = alpha
```
However, `alpha` is passed as `0` from `from_observation` in case there is no background estimate. In `ObservationStats.stack`, there is a check for `a_off > 0`, which would pass here even though it shouldn't

One easy solution would be to add another if clause:
```
if alpha is not None and alpha > 0:
            self.a_on = alpha
            self.a_off = 1
            self.alpha_obs = alpha
```

What was the reason for setting `a_off` to 1 here, anyway?

I can open a PR with the above fix if you want to, but I wanted to hear your thoughts first.

**Comment by cdeil** [comment]
@lmohrmann - I'm not sure.

From looking at it briefly, `ObservationStats`, `ObservationSummary` and `ObservationTableSummary` are small helpers not used from within Gammapy.

The commit that caused the regression for you was done in #1434 which was triggered by #1428. Extending the existing test with an assert on `gamma_rate` would be good before touching the class again.

Maybe it would help to reduce the if / else complexity to remove `alpha` as a parameter in `__init__`, and make it a computed property of `alpha = a_on / a_off` always, just how `excess` is a computed property and not an input to `__init__`?
https://docs.gammapy.org/0.8/api/gammapy.data.ObservationStats.html
You would probably have to adapt one or two callers, but as long as tests pass I would say it's OK.
Note that one can always fill `a_on=alpha, a_off=1` if one only has an alpha computed already for some reason.

Basically, @lmohrmann,  I think whatever PR you send will be OK. Please try to add a few more asserts in `test_obs_stats.py`, especially a  regression test for your current issue.

**Comment by lmohrmann** [comment]
I like the idea of removing `alpha` as a parameter to `__init__`. Any objections to removing `gamma_rate` and `bg_rate` as well? I don't think there are many cases in which one would want to specify these explicitly, and in any case currently none of the other attributes are computed from these rates in case they are specified (so the user may even construct this class with contradicting inputs!). Removing these three parameters would clean up the code quite a bit. I cannot find any places in the code where any of these parameters is passed to `__init__` (except `from_observation` and `stack`, of course, but that's superfluous).

**Comment by cdeil** [comment]
> Any objections to removing gamma_rate and bg_rate as well?

That sounds good to me, please go ahead.

One decision is whether to put default values like `a_on=1`, `a_off=1`,  `livetime="1 second"` ... in `__init__` or not, either having defaults or not seems reasonable to me, feel free to put as you prefer.

One caveat: overall in Gammapy we are tending to remove convenience classes like ObservationStats, ObservationSummary and ObservationTableSummary that don't do much, either completely, or rewrite them as a method on some dataset class that returns a Python dict or Astropy Table with the numbers of interest. In 2019, before v1.0, once the core functionality in Gammapy is better developed, we'll have to go through and discuss what to do with such classes. But for now, it's great if you improve this, there's no concrete plan concerning larger changes.

**Comment by lmohrmann** [comment]
Fixed after #1969
